### PR TITLE
Adds mnemonic phrase derivation scheme

### PR DESCRIPTION
**Pull request by coreyphillips**
This PR:
Adds the mnemonic phrase derivation scheme that will allow the wallet to derive the lightning, tokens and slashtags phrases/seeds from a given (on-chain) mnemonic.
- Created `deriveMnemonicPhrases` method in `utils/wallet/index.ts`.
- Renamed `getMnemonicPhraseFromEntropy` to `generateMnemonicPhraseFromEntropy` in `utils/wallet/index.ts`.
- Updated `getLightningSeed` method in `utils/lightning/index.ts`.
- Created `wallet.ts` tests in `__tests__` directory.
- Added seed derivation test to `__tests__/wallet.ts`.

### CODE REVIEW

**Review by limpbrains — APPROVED** [review]

File: src/utils/lightning/index.ts [hunk 1]
```diff
@@ -61,8 +61,11 @@ export const getLightningSeed = async (
 		if (mnemonic.isErr()) {
 			return err(mnemonic.error.message);
 		}
-		const lndSeed = getMnemonicPhraseFromEntropy(mnemonic.value);
-		return ok(lndSeed);
+		const lndSeed = await deriveMnemonicPhrases(mnemonic.value);
```
Comment (Jasonvdb) [inline src/utils/lightning/index.ts]: Can remove this one completely now, new lightning won't have a seed. Just 32-byte entropy to derive the nodes network pubkey. We can probably just derive that from the main mnemonic later.

**Review by Jasonvdb — APPROVED** [review]

File: src/utils/lightning/index.ts (hunk 1 above)
Comment (coreyphillips) [reply src/utils/lightning/index.ts]: Removed [here](https://github.com/synonymdev/bitkit/pull/152/commits/1df8c379449ba46f4409c00c8cfd52124870e42b).

File: src/utils/wallet/index.ts [hunk 2]
```diff
@@ -476,11 +476,41 @@ export const getMnemonicPhrase = async (
...
+> => {
+	if (!mnemonic) {
+		return err('Please provide a mnemonic phrase.');
+	}
+	const isValid = validateMnemonic(mnemonic);
+	if (!isValid) {
+		return err('Mnemonic provided is not valid.');
+	}
+	const onchain = mnemonic;
+	const lightning = generateMnemonicPhraseFromEntropy(`${mnemonic}lightning`);
```
Comment (instabot42) [inline src/utils/wallet/index.ts:508]: Might it be worth making this `${mnemonic}Synonym.lightning` to add a company level namespace to these. Might reduce clashes in the future

Comment (Jasonvdb) [reply src/utils/wallet/index.ts:508]: Clashes with other wallets? Or if someone forks our wallet and launches their own?

**Pull request merged**